src/main/java/com/linecorp/armeria/server/thrift/ThriftService.java

```diff
@@ -31,25 +31,44 @@
 public class ThriftService extends SimpleService {
 
     /**
-     * Creates a new instance with the specified service implementation and the
-     * {@link ThriftProtocolFactories#BINARY TBinary} protocol.
+     * Creates a new instance with the specified service implementation, supporting all thrift protocols and
+     * defaulting to {@link SerializationFormat#THRIFT_BINARY TBinary} protocol when the client doesn't
+     * specify one. Currently, the only way to specify a serialization format is by using the HTTP session
+     * protocol and setting the Content-Type header to the appropriate {@link SerializationFormat#mimeType}.
      *
      * @param thriftService an implementation of {@code *.Iface} or {@code *.AsyncIface} service interface
      *                      generated by the Apache Thrift compiler
      */
     public ThriftService(Object thriftService) {
-        this(thriftService, ThriftProtocolFactories.BINARY);
+        this(thriftService, SerializationFormat.THRIFT_BINARY);
```

Comment:
How about `this(thriftService, SerializationFormat.THRIFT_BINARTY, SerializationFormat.ofThrift())`?